fs/mmap/fs_mmap.c

```diff
@@ -140,20 +145,6 @@ static int file_mmap_(FAR struct file *filep, FAR void *start,
 
   if (filep->f_inode && filep->f_inode->u.i_ops->mmap != NULL)
     {
-      /* Pass the information about the mapping in mm_map_entry_s structure.
-       * The driver may alter the structure, and if it supports unmap, it
-       * will also add it to the kernel maintained list of mappings.
-       */
-
-      struct mm_map_entry_s map =
-        {
-         NULL, /* sq_entry_t */
-         start, length, offset,
-         prot, flags,
-         NULL, /* priv */
-         NULL  /* munmap */
-        };
-
       ret = filep->f_inode->u.i_ops->mmap(filep, &map);
```

Comment:
the driver who implements mmap need call mm_map_add

Reply:
but the caller who call munmap will report the error now. BTW, the new complex driver may need to clear up the resource in munmap callback.